### PR TITLE
Use CoordsXY/ScreenCoordsXY for viewport*_with_z

### LINKED ISSUE #10268: Refactoring Coordinates Structures: Refactor with_z viewport functions
Part of the refactoring project.

Refactor the following functions to use ScreenCoords and CoordsXY:

- [x] 1. `screen_get_map_xy_with_z` #10238
- [ ] 2. `screen_get_map_xy_quadrant_with_z`
- [ ] 3. `screen_get_map_xy_side_with_z`

### PR DESCRIPTION
**Pull request by tupaschoal**
Closes #10268

### CODE REVIEW

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 1]
```diff
@@ -1377,8 +1377,11 @@ static void sub_6E1F34(
                 {
                     int16_t z = gSceneryCtrlPressZ;
 
-                    screen_get_map_xy_quadrant_with_z(x, y, z, grid_x, grid_y, &cl);
-
+                    auto mapCoords = screen_get_map_xy_quadrant_with_z({ x, y }, z, &cl);
+                    if (!mapCoords)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/TopToolbar.cpp]: You will need to set *grid_x to `LOCATION_NULL` to propagate the problem upwards.

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 2]
```diff
@@ -1556,7 +1559,11 @@ static void sub_6E1F34(
             else
             {
                 int16_t z = gSceneryCtrlPressZ;
-                screen_get_map_xy_side_with_z(x, y, z, grid_x, grid_y, &cl);
+                auto mapCoords = screen_get_map_xy_side_with_z({ x, y }, z, &cl);
+                if (!mapCoords)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/TopToolbar.cpp]: as above

**Comment by tupaschoal** [comment]
Done!

**Comment by duncanspumpkin** [comment]
`goto fail;`

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 3]
```diff
@@ -1377,8 +1377,12 @@ static void sub_6E1F34(
                 {
                     int16_t z = gSceneryCtrlPressZ;
 
-                    screen_get_map_xy_quadrant_with_z(x, y, z, grid_x, grid_y, &cl);
-
+                    auto mapCoords = screen_get_map_xy_quadrant_with_z({ x, y }, z, &cl);
+                    if (!mapCoords)
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TopToolbar.cpp]:
```suggestion
                    if (!mapCoords)
                    {
```

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 4]
```diff
@@ -1377,8 +1377,12 @@ static void sub_6E1F34(
...
+                        *grid_x = LOCATION_NULL;
+                        return;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TopToolbar.cpp]:
```suggestion
                        return;
                    }
```

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 5]
```diff
@@ -1556,7 +1560,12 @@ static void sub_6E1F34(
             else
             {
                 int16_t z = gSceneryCtrlPressZ;
-                screen_get_map_xy_side_with_z(x, y, z, grid_x, grid_y, &cl);
+                auto mapCoords = screen_get_map_xy_side_with_z({ x, y }, z, &cl);
+                if (!mapCoords)
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TopToolbar.cpp]:
```suggestion
                if (!mapCoords)
                {
```

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 6]
```diff
@@ -1556,7 +1560,12 @@ static void sub_6E1F34(
...
+                    *grid_x = LOCATION_NULL;
+                    return;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TopToolbar.cpp]:
```suggestion
                    return;
                }
```

**Pull request merged**